### DISCUSSION TITLE
Bundle size bloated for nearly 60% in v2.29.3

### DISCUSSION THREAD

**Discussion by jsliang** [js-fns/date-fns#3191] [Q&A] [answered]
We noticed that the bundle size of `date-fns` increased by 60kB after bumping from v2.29.2 to v2.29.3.
Screenshot taken from https://bundlephobia.com/package/date-fns@2.29.3:
<img width="1205" alt="Screen Shot 2022-09-26 at 3 25 56 PM" src="https://user-images.githubusercontent.com/681110/192255059-ace9f51c-822d-447b-869d-fd7db3fbcc5e.png">

This is quite a significant bundle size increase for a patch version bump.
Not sure if https://github.com/date-fns/date-fns/pull/3167 introduced the size increase. If it did, is it possible to opt-out of IE11 support to get a smaller bundle size?

**Comment by dzhavat** [comment]
I just tried updating to v2.29.3 and got a bigger final bundle. Had to downgrade to v2.29.2.

Looking at the [changes since v2.29.2](https://github.com/date-fns/date-fns/compare/v2.29.2...v2.29.3), #3167 is the most likely reason for the bundle size increase.

**Comment by kashifshamaz21** [comment]
@leshakoss @tan75 @muruvetri Since the bundlesize bump here is caused specifically due to IE-11 support, could we make this support as an opt-in: maybe date-fns can generate separate outputs - default being non-IE-11 compat (same as how it was on `v2.29.2`) and another output which is IE-11 compat. 
This way, we won't penalise those consumers of date-fns with extra bundle-size who don't really need IE-11 support. And that would make this change backwards compatible (bundlesize-wise).

**Comment by kashifshamaz21** [comment, marked as answer]
Lets continue this conversation in #3208.
